react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  const regExDigit = /[0-9]+/;
+  const regExUppercase = /[A-Z]+/;
+  const regExSpecial = /[!@#$%^&*()]+/;
+  let textColor = 'red';
+  let check = '❌';
+  let text = 'A password is required.';
+
+  if (password.length > 0 && password.length < 8) {
+    text = 'Your password is too short.';
+  } else if (password.length > 7 && !regExDigit.test(password)) {
+    text = 'Must contain a number!';
+  } else if (password.length > 7 && !regExUppercase.test(password)) {
+    text = 'Must contain a capital letter!';
+  } else if (password.length > 7 && !regExSpecial.test(password)) {
+    text = 'Must contain a special character "!, @, #, $, %, ^, &, *, (, or )';
+  } else if (password.length > 7) {
+    text = "That's a strong password! 💪";
+    check = '✔️';
+    textColor = 'green';
+  }
+
+  return (
+    <form>
+      <label style={{ display: "block"}} htmlFor="pw">Password</label>
```

Comment:
This is OK, but the best way to use `<label>` is to wrap it around the `<input>` component. Then you don't need `htmlFor` and the input doesn't need `id`.